oak/server/enclave_server.h

```diff
@@ -139,61 +141,33 @@ class EnclaveServer final : public asylo::TrustedApplication {
     }
   }
 
-  // Creates a new gRPC event tag based on a number. gRPC tags are usually employed to uniquely
-  // identify events, but we do not need to keep track of them, and therefore we just create them
-  // based on arbitrary numbers.
-  // See https://grpc.io/grpc/cpp/classgrpc_1_1_completion_queue.html.
-  static void *tag(int i) { return reinterpret_cast<void *>(i); }
-
-  // Processes the next available event from the completion queue, blocking if none is available.
-  void ProcessNextEvent() {
-    void *out_tag;
-    bool ok = false;
-    completion_queue_->Next(&out_tag, &ok);
-    // TODO: Return the value of ok so that we can break the completion queue loop.
-    if (!ok) {
-      LOG(INFO) << "Could not process gRPC event from completion queue";
-    }
+  void RequestNextCall() {
+    // The stream will delete itself when it finishes.
+    auto stream = std::make_shared<grpc_server::GrpcStream>(node_);
```

Comment:
Should we keep a `stream` object around as a class member instead of creating one each time perhaps?

Reply:
The issue is that a lot of the stream content needs to be reset for each request (see assertion being hit https://github.com/grpc/grpc/blob/master/src/cpp/server/server_context.cc#L272).

The node is the only thing that stays across stream requests.

The stream is essentially holding all of the state for an RPC and there can be multiple RPCs at once so it can't be a member of the class.

Reply:
Ah I see; in that case, it seems like in theory each event handler could just keep around whatever gRPC state it needs (e.g. request / response buffers, etc), rather than passing around this entire object to all the handlers? But I see the convenience of encapsulating all the gRPC state in the same place, so up to you really.

Reply:
Yah, I think for request / response buffers it could make sense because they are used in subsequent stages. I didn't do this because I'm not sure if it will be useful to reuse it if we add support for streaming requests

For things like node, it is a bit weird because you need to pass through from the first stage to a later stage and in that case it makes sense to have a context- like object like the one here.